packages/databyss-ui/Editor/EditorPage.js

```diff
@@ -40,18 +42,27 @@ const EditorPage = ({ children }) => {
   }
 
   const onBlockBlur = (id, text, editableState) => {
-    // check if block is empty
     // if empty replace block with virgin block
     if (text.length === 0 && id) {
-      // check block to see if ranges exist
-      dispatchEditor(clearBlock(id, editableState))
+      if (!isBlockEmpty(id, editableState)) {
+        dispatchEditor(clearBlock(id, editableState))
+      }
     }
 
     if (text.trim().match(/^@/) && editorState.blocks[id].type !== 'SOURCE') {
-      onSetBlockType('SOURCE', id, editableState)
+      // if only @ or # was entered
+      if (text.trim().length === 1) {
```

Comment:
As this condition and action is the same for more than one block type, coalesce into one condition (if match @ or # and length is 1)